activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -126,14 +126,19 @@ def migration_context # :nodoc:
       def schema_migration # :nodoc:
         @schema_migration ||= begin
                                 conn = self
-                                spec_name = conn.pool.spec.name
+                                spec_name = conn.pool.db_config.spec_name
                                 name = "#{spec_name}::SchemaMigration"
 
                                 Class.new(ActiveRecord::SchemaMigration) do
                                   define_singleton_method(:name) { name }
                                   define_singleton_method(:to_s) { name }
 
-                                  self.connection_specification_name = spec_name
+                                  connection_handler.connection_pool_names.each do |name|
+                                    if conn.pool == connection_handler.retrieve_connection_pool(name)
+                                      self.connection_specification_name = name
```

Comment:
This is the main weirdness introduced by this refactor, since the `spec_name` is only available in `ConnectionSpecification` I'm forced to go through some inelegant hoops to properly set the `connection_specification_name`.

However I believe this is a just a temporary hack, and that it will be easy to clean it up once `ConnectionSpecification` is entirely gone.